### ISSUE TITLE
Private properties not supported

### ISSUE THREAD

**Issue by ttfkam** [eslint/eslint#12680]
**Tell us about your environment**

* **ESLint Version:** 6.7.2
* **Node Version:** 12.13.0
* **npm Version:** 6.12.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
Basic file with Node 12 on MacOS Catalina, though acts the same in a Docker container.
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
class IncreasingCounter {
  #count = 0;
  get value() {
    console.log('Getting the current value!');
    return this.#count;
  }
  increment() {
    this.#count++;
  }
}
```

```bash
npx eslint .
```

**What did you expect to happen?**
Parse current JavaScript

**What actually happened? Please include the actual, raw output from ESLint.**
Complains about the '#' character.

**Are you willing to submit a pull request to fix this bug?**

**Comment by ilyavolodin** [comment]
Class fields proposal is still Stage 3. ESLint does not support language features that haven't reached Stage 4. We will support it once second engine implements it (first one being Chrome) and proposal moves into Stage 4.

**Comment by kaicataldo** [comment]
Closing this as it's not currently actionable.
